src/frequenz/client/microgrid/_component_states.py

```diff
@@ -6,7 +6,102 @@
 from enum import Enum
 from typing import Self
 
-from frequenz.microgrid.betterproto.frequenz.api.microgrid import ev_charger
+from frequenz.microgrid.betterproto.frequenz.api.microgrid import (
+    battery,
+    ev_charger,
+    inverter,
+)
+
+
+class BatteryComponentState(Enum):
+    """Component states of a battery."""
+
+    UNSPECIFIED = battery.ComponentState.COMPONENT_STATE_UNSPECIFIED
+    """Unspecified component state."""
+
+    OFF = battery.ComponentState.COMPONENT_STATE_OFF
+    """The battery is switched off."""
+
+    IDLE = battery.ComponentState.COMPONENT_STATE_IDLE
+    """The battery is idle."""
+
+    CHARGING = battery.ComponentState.COMPONENT_STATE_CHARGING
+    """The battery is consuming electrical energy."""
+
+    DISCHARGING = battery.ComponentState.COMPONENT_STATE_DISCHARGING
+    """The battery is generating electrical energy."""
+
+    ERROR = battery.ComponentState.COMPONENT_STATE_ERROR
+    """The battery is in a faulty state."""
+
+    LOCKED = battery.ComponentState.COMPONENT_STATE_LOCKED
+    """The battery is online, but currently unavailable.
+
+    Possibly due to a pre-scheduled maintenance, or waiting for a resource to be loaded.
+    """
+
+    SWITCHING_ON = battery.ComponentState.COMPONENT_STATE_SWITCHING_ON
+    """
+    The battery is starting up and needs some time to become fully operational.
+    """
+
+    SWITCHING_OFF = battery.ComponentState.COMPONENT_STATE_SWITCHING_OFF
+    """The battery is switching off and needs some time to fully shut down."""
+
+    UNKNOWN = battery.ComponentState.COMPONENT_STATE_UNKNOWN
+    """The state is provided by the device, but it is not one of the above states."""
```

Comment:
I think we should say component, not device.

```suggestion
    """A state is provided by the component, but it is not one of the above states."""